Streamline exception handling

HazAT opened the pull request:
This streamlines the exception handling and always uses raven-js to capture the exception, we just use the native client as a transport layer.

- Fix #110

- This is required https://github.com/getsentry/sentry-java/pull/428

vafada commented:
@HazAT 

If you just use native code as a transport layer then what's the advantage of using native code vs using raven-js to send message/data to sentry server?

also does this change capture native errors/stacktrace (not JS errors, Java or Objective-C error) and report to sentry server?

HazAT commented:
Hey, we recommend using the native code as a transport layer because you get a lot more useful information out of your crashes. Mostly information about the device.
I've updated the readme and added more screenshots why it's better :)
